smithy-model/src/main/java/software/amazon/smithy/model/traits/EnumValueTrait.java

```diff
@@ -97,13 +96,12 @@ public Provider() {
         @Override
         public Trait createTrait(ShapeId target, Node value) {
             Builder builder = builder().sourceLocation(value);
-            ObjectNode objectNode = value.expectObjectNode();
-            objectNode.getMember("string")
-                    .map(v -> v.expectStringNode().getValue())
-                    .ifPresent(builder::stringValue);
-            objectNode.getMember("int")
-                    .map(v -> v.expectNumberNode().getValue().intValue())
-                    .ifPresent(builder::intValue);
+            value.asStringNode().ifPresent(node -> builder.stringValue(node.getValue()));
+            value.asNumberNode().ifPresent(node -> {
+                if (node.isNaturalNumber()) {
+                    builder.intValue(node.getValue().intValue());
+                }
```

Comment:
Shouldn't this have an `else` that throws?

Reply:
The constructor will throw if neither is set, but this could probably use a more specific error.

Reply:
If the number has a fractional part, then the constructor will throw saying no value was provided? Ideally it would say we don't support fractional precision right? Only integers.